src/components/Toast/Toast.tsx

```diff
@@ -1,93 +1,70 @@
+   * Note that the Icon mapping matches the style of Banners.
    */
-  iconTitle?: string;
+  variant: Variant;
   /**
-   * Stylistic variations for the toast type.
-   * - **success** - results in a green toast
-   * - **warning** - results in a yellow toast
-   * - **error** - results in a red toast
-   * - **info** - results in a blue toast
+   * Callback when Toast is dismissed.
    */
-  variant?: 'success' | 'warning' | 'error' | 'info' | 'brand';
-}
+  onDismiss?: () => void;
+};
 
 /**
- * Primary UI component for user interaction
+ * ```ts
+ * import {Toast} from "@chanzuckerberg/eds-components";
+ * ```
+ *
+ * A toast used to provide information on the state of the page, usually in response to a
+ * user action. Ex: The user updates their profile, and a toast pop-up informs them that the
+ * data was successfully saved.
  */
-export const Toast: React.FC<Props> = ({
-  iconTitle,
-  className,
+export const Toast = ({
   children,
-  iconName,
-  dismissible = true,
+  className,
   variant,
-  closeButtonText = 'Close',
+  onDismiss,
+  // Allow for additional attributes such as aria roles
   ...other
-}) => {
+}: Props) => {
   const componentClassName = clsx(
-    styles['toast'],
     className,
-    variant === 'success' && styles['toast--success'],
-    variant === 'warning' && styles['toast--warning'],
-    variant === 'error' && styles['toast--error'],
-    variant === 'brand' && styles['toast--brand'],
+    styles.toast,
+    /* TODO: uncomment and point to corresponding style classes once Notifications is fully migrated */
+    // variant === 'success' && colorStyles.colorSuccess,
+    // variant === 'alert' && colorStyles.colorAlert,
   );
-  const [dismissed, setDismissed] = useState(false);
-
-  function onDismiss(e: any) {
-    e.preventDefault();
-    setDismissed(true);
-  }
-  return dismissed ? null : (
-    <div className={componentClassName} role="alert" {...other}>
-      {iconName && (
-        <Icon
-          name={iconName}
-          title={iconTitle}
-          className={styles['toast__icon']}
-        />
-      )}
-      <div className={styles['toast__body']}>{children}</div>
-
-      {dismissible && (
-        <Button
-          className={styles['toast__close-btn']}
-          variant="bare"
-          aria-label={closeButtonText}
-          iconName="x"
-          iconPosition="after"
-          onClick={(e: any) => onDismiss(e)}
-        />
-      )}
+  return (
+    <div className={componentClassName} {...other}>
+      <div className={styles['toast--content']}>
+        {/* TODO: point to internal NotificationIcon once fully migrated */}
+        {/* <NotificationIcon variant={variant} /> */}
+        <p className={styles['toast--text']}>{children}</p>
+      </div>
+      {/* TODO: point to internal CloseButton once fully migrated */}
+      {/* {onDismiss && <CloseButton color={variant} onClose={onDismiss} />} */}
+      {onDismiss && <button onClick={onDismiss}>temp close tooltip</button>}
```

Comment:
I know you saw this in the other PR but I'll leave a reminder here just in case that we're moving away from the `CloseButton`. Maybe use the "bare" `Button` variant for now and change the comment to update this button when the variants have been updated.

Reply:
Sounds good, will use `button`